### PR TITLE
update cluster logs and test artifacts during test recording

### PR DESCRIPTION
**Pull request by tianleh**
Signed-off-by: Tianle Huang <tianleh@amazon.com>

### Description
This PR will copy Dashboards Cypress test artifacts into the persistent dir and also save OpenSearch logs when running Dashboards tests (for ease of debugging)

Used `shutil.copytree` instead of `shutil.copyfile` for
1) simplicity
2) Cypress test folders have hierarchies
 
### Issues Resolved
Resolves https://github.com/opensearch-project/opensearch-build/issues/1190
Resolves https://github.com/opensearch-project/opensearch-build/issues/1191

### Test
1 unit test in progress
2 manual test
Verified that the logs are saved.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianleh** [comment]
Send the functional part for early feedback. Unit test is in progress.

File: src/test_workflow/integ_test/integ_test_suite.py [hunk 1]
```diff
@@ -88,6 +87,10 @@ def pretty_print_message(self, message):
         logging.info(message)
         logging.info("===============================================")
 
+    @abc.abstractmethod
```
Comment (dblock) [inline src/test_workflow/integ_test/integ_test_suite.py:90]: Make it a property called `test_artifact_files`.

File: src/test_workflow/integ_test/service.py [hunk 2]
```diff
@@ -103,6 +99,10 @@ def wait_for_service(self):
             time.sleep(10)
         raise ClusterCreationException("Cluster is not available after 10 attempts")
 
+    @abc.abstractmethod
+    def get_log_files(self):
```
Comment (dblock) [inline src/test_workflow/integ_test/service.py:103]: Property called `log_files`

File: src/test_workflow/test_recorder/test_recorder.py [hunk 3]
```diff
@@ -48,6 +48,13 @@ def _generate_yml(self, test_result_data: TestResultData, output_path):
             yaml.dump(outcome, file)
         return os.path.realpath("%s.yml" % test_result_data.component_name)
 
+    def _copy_log_files(self, log_files, dest_directory):
+        if log_files is not None:
```
Comment (dblock) [inline src/test_workflow/test_recorder/test_recorder.py:52]: remove unnecessary `is not None`

**Review by dblock — APPROVED** [review]
Some renaming and minor nits, otherwise LGTM.

**Review by seraphjiang — APPROVED** [review]

**Pull request merged**